src/main/java/edu/hm/hafner/analysis/IssueFilterConfig.java

```diff
@@ -0,0 +1,279 @@
+package edu.hm.hafner.analysis;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * A config for filtering issues. Filter-options are FileName, PackageName, ModuleName, Category and Type. For each of
+ * them there is a possibility to include and exclude issues. The filters are regexes. Instantiation only with
+ * IssueFilterConfigBuilder.
+ */
+public class IssueFilterConfig {
+    private final boolean isIncludeSet;
+
+    private final Pattern includeFileName;
+    private final Pattern includePackageName;
+    private final Pattern includeModuleName;
+    private final Pattern includeCategory;
+    private final Pattern includeType;
+
+    private final Pattern excludeFileName;
+    private final Pattern excludePackageName;
+    private final Pattern excludeModuleName;
+    private final Pattern excludeCategory;
+    private final Pattern excludeType;
+
+    /**
+     * Ctor for Builder.
+     *
+     * @param builder
+     *         contains the filter-parameters.
+     */
+    private IssueFilterConfig(IssueFilterConfigBuilder builder) {
+        this.isIncludeSet = builder.isIncludePatternSet;
+        this.includeFileName = builder.includeFileName;
+        this.includePackageName = builder.includePackageName;
+        this.includeModuleName = builder.includeModuleName;
+        this.includeCategory = builder.includeCategory;
+        this.includeType = builder.includeType;
+        this.excludeFileName = builder.excludeFileName;
+        this.excludePackageName = builder.excludePackageName;
+        this.excludeModuleName = builder.excludeModuleName;
+        this.excludeCategory = builder.excludeCategory;
+        this.excludeType = builder.excludeType;
+    }
+
+    public boolean isIncludePatternSet() {
+        return isIncludeSet;
+    }
+
+    public Pattern getIncludeFileName() {
```

Comment:
Bei den include-Gettern könnte gleich ein Predicate zurückgegeben werden.
Dann würden die Aufrufe in `Issues.filterIssues` kürzer sein.
`
public Predicate<String> getIncludeFileName() {
    return includeFileName.asPredicate();
}
`